tgui/packages/tgui/interfaces/Fabricator.tsx

```diff
@@ -0,0 +1,337 @@
+import { useBackend, useSharedState } from '../backend';
+import { Stack, Section, Button, Input, Icon, Tabs, Dimmer } from '../components';
+import { Window } from '../layouts';
+import { Material, MaterialAmount, MaterialFormatting, Materials, MATERIAL_KEYS } from './common/Materials';
+import { Fragment } from 'inferno';
+
+type Design = {
+  name: string;
+  desc: string;
+  cost: { [K in keyof typeof MATERIAL_KEYS]: number };
+  id: string;
+  categories?: string[];
+  maxstack: number;
+  icon: string;
+};
+
+type FabricatorData = {
+  materials: Material[];
+  fab_name: string;
+  on_hold: boolean;
+  designs: { [K: string]: Design };
+  busy: boolean;
+};
+
+type AvailableMaterials = { [K in keyof typeof MATERIAL_KEYS]?: number };
+
+const SearchBar = (props, context) => {
+  const [searchText, setSearchText] = useSharedState(
+    context,
+    'search_text',
+    ''
+  );
+
+  return (
+    <Stack align="baseline">
+      <Stack.Item>
+        <Icon name="search" />
+      </Stack.Item>
+      <Stack.Item grow>
+        <Input
+          fluid
+          placeholder="Search for..."
+          onInput={(_e: unknown, v: string) => setSearchText(v.toLowerCase())}
```

Comment:
You shouldn't need the types here and they don't help much with clarity, Ideally `onInput` would pass them in, but if Input isn't TypeScript yet, then this should work anyway.

Reply:
Yeah, `onInput` is still `(any, any) => any`, so I manually put types here.